src/Network/Email/Email.php

```diff
@@ -1885,4 +1890,78 @@ protected function _getContentTypeCharset()
         }
         return strtoupper($this->charset);
     }
+
+    /**
+     * Serializes the email object to a value that can be natively serialized and re-used
+     * to clone this email instance.
+     *
+     * It has certain limitations for viewVars that are good to know:
+     *
+     *    - ORM\Query executed and stored as resultset
+     *    - File resources read and stored as contents
+     *    - SimpleXmlElements stored as associative array
+     *    - Exceptions stored as strings
+     *    - Closures and PDOs are not supported.
+     *
+     * @return array Serializable array of configuration properties.
+     * @throws \Exception When a view var object can not be properly serialized.
+     */
+    public function jsonSerialize()
+    {
+        $properties = [
+            '_to', '_from', '_sender', '_replyTo', '_cc', '_bcc', '_subject', '_returnPath', '_readReceipt',
+            '_template', '_layout', '_viewRender', '_viewVars', '_theme', '_helpers', '_emailFormat',
+            '_emailPattern', '_attachments', '_domain', '_messageId', '_headers', 'charset', 'headerCharset',
+        ];
+
+        $array = [];
+
+        foreach ($properties as $property) {
+            $array[$property] = $this->{$property};
+        }
+
+        // Iterate through view vars to clean up and normalize
+        array_walk_recursive($array['_viewVars'], function (&$item, $key) {
+            // Store file contents instead of resource.
+            if (is_resource($item) && get_resource_type($item) == 'stream') {
+                $item = stream_get_contents($item);
+            }
```

Comment:
Looking at code of `Email::attachments()` even those won't have resources. Each attachment in `Email::$_attachments` is represented by an array which can either have a `file` key which is path to file or `data` key which has base64 encoded and chunk splitted content which is already good for serializing. So this loop needs to be changed to also read the items which have paths and base64 encoded and chunk split their content.